File: autods/service/report/job_cards_by_status/job_cards_by_status.js
// Copyright (c) 2026, Agilasoft Technologies Inc. and contributors
// For license information, please see license.txt

frappe.query_reports["Job Cards by Status"] = {
	"filters": [
		{"fieldname": "from_date", "label": __("From Date"), "fieldtype": "Date", "default": frappe.datetime.month_start()},
		{"fieldname": "to_date", "label": __("To Date"), "fieldtype": "Date", "default": frappe.datetime.month_end()},
		{"fieldname": "work_area", "label": __("Work Area"), "fieldtype": "Link", "options": "Work Area"},
		{"fieldname": "status", "label": __("Status"), "fieldtype": "Select", "options": ["", "Open", "Work In Progress", "On Hold", "Completed", "Cancelled"]}
	]
};

File: autods/service/report/job_cards_by_status/job_cards_by_status.py
# Copyright (c) 2026, Agilasoft Technologies Inc. and contributors
# For license information, please see license.txt

import frappe
from frappe import _


def execute(filters=None):
	columns = get_columns()
	data = get_data(filters)
	return columns, data


def get_columns():
	return [
		{"fieldname": "status", "label": _("Status"), "fieldtype": "Data", "width": 140},
		{"fieldname": "work_area", "label": _("Work Area"), "fieldtype": "Link", "options": "Work Area", "width": 120},
		{"fieldname": "job_count", "label": _("Job Cards"), "fieldtype": "Int", "width": 100},
	]


def get_data(filters):
	conditions = ["j.docstatus >= 0"]
	values = {}
	if filters.get("from_date"):
		conditions.append("j.repair_date >= %(from_date)s")
		values["from_date"] = filters["from_date"]
	if filters.get("to_date"):
		conditions.append("j.repair_date <= %(to_date)s")
		values["to_date"] = filters["to_date"]
	if filters.get("work_area"):
		conditions.append("j.work_area = %(work_area)s")
		values["work_area"] = filters["work_area"]
	if filters.get("status"):
		conditions.append("j.status = %(status)s")
		values["status"] = filters["status"]
	where = " and ".join(conditions)
	query = f"""
		select
			j.status,
			j.work_area,
			count(*) as job_count
		from `tabJob Card` j
		where {where}
		group by j.status, j.work_area
		order by j.status, j.work_area
	"""
	return frappe.db.sql(query, values, as_dict=1)
